src/pages/settings/Profile/PersonalDetails/DateOfBirthPage.js

```diff
@@ -38,6 +38,40 @@ class DateOfBirthPage extends Component {
 
         this.validate = this.validate.bind(this);
         this.updateDateOfBirth = this.updateDateOfBirth.bind(this);
+        this.readParams = this.readParams.bind(this);
+        this.state = {
+            defaultMonth: undefined,
+            defaultYear: undefined,
+        };
+
+        this.onDateChanged = this.onDateChanged.bind(this);
+    }
+
+    componentDidMount() {
+        this.props.navigation.addListener('focus', this.readParams);
```

Comment:
@s77rt In this case, adding the withNavigation higher-order component (HOC) is unnecessary because the DateOfBirthPage is the first component rendered within a StackNavigator.

The navigation prop is automatically passed down to the component as a prop by the StackNavigator. This allows the component to access the navigation object and its methods without needing the withNavigation HOC.

Reply:
@ArekChr Makes sense! Thanks!